pytensor/link/jax/ops.py

```diff
@@ -0,0 +1,472 @@
+            if isinstance(output_gradients[i].type, DisconnectedType):
+                zero_shape = (
+                    self.output_types[i].shape
+                    if None not in self.output_types[i].shape
+                    else ()
+                )
+                output_gradients[i] = pt.zeros(zero_shape, self.output_types[i].dtype)
+
+        # Compute the gradient.
+        grad_result = self.vjp_jax_op(inputs, output_gradients)
+        return grad_result if self.num_inputs > 1 else (grad_result,)
+
+
+class VJPJAXOp(Op):
+    def __init__(self, input_types, jitted_vjp, name=None):
+        self.input_types = input_types
+        self.jitted_vjp = jitted_vjp
+        if name is not None:
+            self.custom_name = name
+            self.__class__.__name__ = name
+            self.__class__.__qualname__ = ".".join(
+                self.__class__.__qualname__.split(".")[:-1] + [name]
+            )
+
+    def make_node(self, y0, gz):
+        y0_converted = [
+            pt.as_tensor_variable(y).astype(self.input_types[i].dtype)
+            for i, y in enumerate(y0)
+        ]
+        gz_not_disconnected = [
+            pt.as_tensor_variable(g)
+            for g in gz
+            if not isinstance(g.type, DisconnectedType)
+        ]
+        outputs = [typ() for typ in self.input_types]
+        self.num_outputs = len(outputs)
+        return Apply(self, y0_converted + gz_not_disconnected, outputs)
+
+    def perform(self, node, inputs, outputs):
+        results = self.jitted_vjp(*inputs)
+        if len(self.input_types) > 1:
+            for i, res in enumerate(results):
+                outputs[i][0] = np.array(res, self.input_types[i].dtype)
+        else:
+            outputs[0][0] = np.array(results, self.input_types[0].dtype)
+
+    def perform_jax(self, *inputs):
+        return self.jitted_vjp(*inputs)
+
+
+def _normalize_flat_func(func):
+    def normalized_func(*flat_vars):
+        out_flat = func(*flat_vars)
+        if isinstance(out_flat, Sequence):
+            return tuple(out_flat) if len(out_flat) > 1 else out_flat[0]
+        else:
+            return out_flat
+
+    return normalized_func
+
+
+def _get_vjp_jax_op(flat_func, num_out):
+    def vjp_op(*args):
+        y0 = args[:-num_out]
+        gz = args[-num_out:]
+        if len(gz) == 1:
+            gz = gz[0]
+
+        def f(*inputs):
+            return flat_func(*inputs)
+
+        primals, vjp_fn = jax.vjp(f, *y0)
+
+        def broadcast_to_shape(g, shape):
+            if g.ndim > 0 and g.shape[0] == 1:
+                g_squeezed = jnp.squeeze(g, axis=0)
+            else:
+                g_squeezed = g
+            return jnp.broadcast_to(g_squeezed, shape)
+
+        gz = tree_map(
+            lambda g, p: broadcast_to_shape(g, jnp.shape(p)).astype(p.dtype),
+            gz,
+            primals,
+        )
+        return vjp_fn(gz)
+
+    return vjp_op
+
+
+def as_jax_op(jaxfunc, use_infer_static_shape=True, name=None):
```

Comment:
Allowing passing input_types, output_types that are forwarded to the init of the Op. Optionally